### DISCUSSION TITLE
RabbitMQ Shutdown: Reason: noproc | Reason: shutdown

### DISCUSSION THREAD

**Discussion by Pramod34** [rabbitmq/rabbitmq-server#10294] [Other]
We have an applications that is RabbitMQ which is getting continuous data, and our applications seems to be running fine. But we look into the RabbitMQ logs we noticed few errors.

1. "client unexpectedly closed TCP connection"
2. "supervior: {<0.20773.10>, rabbit_channel_sup}; errorContext: shutdown_error; reason: noproc"
3. "supervior: {<0.19363.10>, rabbit_channel_sup_sup}; errorContext: shutdown_error; reason: shutdown"

Can you please help me understand these issues and fix them? We are seeing these issues in our PROD environment.

RabbitMQ version: 3.11.9
Erlang version: 25.3.2

I really thankful to your help.

**Comment by lukebakken** [comment]
> "client unexpectedly closed TCP connection"

Your application is not gracefully closing the AMQP connection to RabbitMQ, or a network device in-between is closing the connection.

The other two messages may be related to the above, but without more context it is hard to know.

**Reply by Pramod34** [reply]
Would that issue be an impact to production? What context do we need to provide to debug more details for the cause of the shutdown error issue?

**Reply by lukebakken** [reply]
You can ignore all of these messages, they are just log noise.

Please carefully read the document to which @michaelklishin linked in [his comment](https://github.com/rabbitmq/rabbitmq-server/discussions/10294#discussioncomment-8057423).

**Reply by lukebakken** [reply]
If you want to fix the `client unexpectedly closed TCP connection` log messages, your applications will have to be modified so that they close their connections gracefully.

**Comment by michaelklishin** [comment]
1. This message has an [almost dedicated doc section](https://rabbitmq.com/logging.html#logged-events)
